### PR TITLE
feat: Update button style to new Elm Theme

### PR DESCRIPTION
**Pull request by omerhabib26**
### Description:

- Define generic button style for the whole app
- Remove the custom/random button initialization
- Make dialog's buttons appearance close to native

fix: LEARNER-10083 + LEARNER-10084

Ref: https://www.figma.com/design/q6z6cpJsbBBcnPZLPS7KEu/Marketplace-Visual-Design?node-id=3932-2331&t=C9RFjYFEPRxwC2Hc-0

Screenshot 2024-09-03 at 7 48 34 PM
Screenshot 2024-09-03 at 7 48 44 PM
Screenshot 2024-09-03 at 7 48 55 PM
Screenshot 2024-09-03 at 7 49 06 PM
Screenshot 2024-09-03 at 7 49 14 PM
Screenshot 2024-09-03 at 7 49 26 PM
Screenshot 2024-09-03 at 7 49 34 PM
Screenshot 2024-09-03 at 7 49 51 PM
Screenshot 2024-09-03 at 7 49 59 PM
Screenshot 2024-09-03 at 7 50 09 PM
Screenshot 2024-09-03 at 7 50 36 PM
Screenshot 2024-09-03 at 7 50 59 PM
Screenshot 2024-09-03 at 7 51 15 PM
Screenshot 2024-09-03 at 7 52 20 PM
Screenshot 2024-10-09 at 6 48 45 PM
Screenshot 2024-10-09 at 6 48 53 PM
Screenshot 2024-10-09 at 6 51 45 PM
Screenshot 2024-10-09 at 6 51 58 PM

### CODE REVIEW

**Review by dixidroid — CHANGES_REQUESTED** [review]

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]
Need product approval for the design changes on the following files:

- Fullscreen(error message).
- ChapterEndDialogScreen
- SetupCourseAccessErrorButtons
- CourseOutlineScreen
- NotSupportedUnitFragment
- DashboardGalleryView(Find new course button)
- DiscussionThreadsFragment
- CalendarAccessDialogFragment

File: course/src/main/java/org/openedx/course/presentation/ui/CourseVideosUI.kt [hunk 1]
```diff
@@ -204,7 +204,7 @@ private fun CourseVideosUI(
         }
 
         var isDownloadConfirmationShowed by rememberSaveable {
-            mutableStateOf(false)
+            mutableStateOf(true)
```
[the remark this answers is not in the document]
Comment (omerhabib26) [reply course/src/main/java/org/openedx/course/presentation/ui/CourseVideosUI.kt:207]: updated by mistake, reverted.

File: core/src/main/java/org/openedx/core/ui/ComposeCommon.kt [hunk 2]
```diff
@@ -1071,22 +1075,28 @@ fun OpenEdXButton(
...
+            .heightIn(min = 42.dp),
         shape = MaterialTheme.appShapes.buttonShape,
         colors = ButtonDefaults.buttonColors(
-            backgroundColor = backgroundColor
+            backgroundColor = backgroundColor,
+            disabledBackgroundColor = backgroundColor.copy(alpha = 0.3f)
         ),
         enabled = enabled,
-        onClick = onClick
+        onClick = onClick,
```
[the remark this answers is not in the document]
Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/ui/ComposeCommon.kt:1091]: decided to use trailing commas.

File: core/src/main/java/org/openedx/core/ui/ComposeCommon.kt [hunk 3]
```diff
@@ -1095,33 +1105,121 @@ fun OpenEdXButton(
...
     modifier: Modifier = Modifier.fillMaxWidth(),
-    backgroundColor: Color = Color.Transparent,
-    borderColor: Color,
-    textColor: Color,
     text: String = "",
-    enabled: Boolean = true,
     onClick: () -> Unit,
+    enabled: Boolean = true,
+    textColor: Color = MaterialTheme.appColors.primaryButtonBackground,
+    backgroundColor: Color = Color.White,
```
[the remark this answers is not in the document]
Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/ui/ComposeCommon.kt:1115]: still pending, we can cover this with Standardized Styling System work.

File: core/src/main/java/org/openedx/core/ui/ComposeCommon.kt [hunk 4]
```diff
@@ -1071,22 +1075,28 @@ fun OpenEdXButton(
...
         modifier = Modifier
-            .testTag("btn_${text.tagId()}")
+            .testTag(
+                text
+                    .takeIfNotEmpty()
+                    ?.let { "btn_${text.tagId()}" } ?: ""
+            )
             .then(modifier)
-            .height(42.dp),
+            .heightIn(min = 42.dp),
```
[the remark this answers is not in the document]
Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/ui/ComposeCommon.kt:1084]: still pending.

File: dashboard/src/main/java/org/openedx/courses/presentation/DashboardGalleryView.kt [hunk 5]
```diff
@@ -944,19 +944,14 @@ private fun FindACourseButton(
     modifier: Modifier = Modifier,
     findACourseClick: () -> Unit
 ) {
-    OpenEdXButton(
+    BrandButton(
         modifier = modifier
-            .fillMaxWidth()
```
[the remark this answers is not in the document]
Comment (farhan-arshad-dev) [reply dashboard/src/main/java/org/openedx/courses/presentation/DashboardGalleryView.kt:949]: still pending.

**Comment by farhan-arshad-dev** [comment]
> Need product approval for the design changes on the following files:
> 
> * Fullscreen(error message).
> * ChapterEndDialogScreen
> * SetupCourseAccessErrorButtons
> * CourseOutlineScreen
> * NotSupportedUnitFragment
> * DashboardGalleryView(Find new course button)
> * DiscussionThreadsFragment
> * CalendarAccessDialogFragment

Have we received product approval for these changes?

Also need to address 2 comments, other than that ready to be approved.

**Review by farhan-arshad-dev — APPROVED** [review]

**Review by farhan-arshad-dev — APPROVED** [review]

**Review by dixidroid — APPROVED** [review]

**Pull request merged**